### PR TITLE
Op version python mkldnn_inplace test 

### PR DESCRIPTION
**Pull request by lidanqing-vv**
### PR types
Others 

### PR changes
Others

### Describe
Add python mkldnn_inplace function test and op version test

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/mkldnn_inplace_pass.cc [hunk 1]
```diff
@@ -17,10 +17,12 @@
 #include <memory>
 #include <string>
 #include <unordered_map>
+#include <unordered_set>
```
Comment (lidanqing-vv) [inline paddle/fluid/framework/ir/mkldnn/mkldnn_inplace_pass.cc:20]: This unordered set is required by pre-commit when I commit changes
```
paddle/fluid/framework/ir/mkldnn/mkldnn_inplace_pass.cc:199:  Add #include <unordered_set> for unordered_set<>  [build/include_what_you_use] [4]
Done processing paddle/fluid/framework/ir/mkldnn/mkldnn_inplace_pass.cc
```

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_inplace_fuse_pass.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                data, num_filters=3, filter_size=3, bias_attr=False)
+            softmax_out = fluid.layers.softmax(conv_out)
+        self.pass_name = 'mkldnn_inplace_pass'
+        self.feeds = {
+            "data": np.random.random((1, 3, 100, 100)).astype("float32")
+        }
+        self.fetch_list = [conv_out, softmax_out]
+        self.enable_mkldnn = True
+        assert conv_out == softmax_out
+        assert conv_out.var == softmax_out.var
```
Comment (lidanqing-vv) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_inplace_fuse_pass.py]: Hi, @juncaipeng  the oneDNN softmax inplace idea is reusing the input variable of softmax and put the output in the same variable.
Do you think this two assert is enough to check that input and output of softmax are using same Variable ?

**Review by wojtuss — APPROVED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@luotao1 这个PR 可以合了吗

**Pull request merged**